user/tests/tests_view.py

```diff
@@ -0,0 +1,114 @@
+# from django.test import TestCase
+from django.test import TestCase
+from user.models import User
+from django.urls import reverse, reverse_lazy
+import json
+from .tests_config import TEST_USER_CREDENTIALS as test_user
+from django.forms.models import model_to_dict
+
+MODEL_FIELDS = ['id', 'username', 'email', 'enable_notifications', 'first_name', 'last_name']
+
+# Create your tests here.
+
+
+class UserViewTest(TestCase):
+    @classmethod
+    def create_user(cls, username=test_user['username'],
+                    email=test_user['email'], first_name='event',
+                    password=test_user['password'], last_name='manager'):
+        user = User.objects.create(
+            username=username, email=email, first_name=first_name, last_name=last_name)
+        user.set_password(password)
+        user.save()
+        return user
+
+    def test_user_list_view(self):
+        user_instance = self.create_user()
+        url = reverse_lazy('user-list')
+        resp = self.client.get(url)
+        self.assertEqual(resp.status_code, 200)
+        json_resp = json.loads(resp.content)['results'].pop()
+        self.assertDictContainsSubset(model_to_dict(user_instance, MODEL_FIELDS), json_resp, 'it should list all users')
+
+    def test_user_detail_view(self):
+        user_instance = self.create_user()
+        url = reverse('user-detail', args=[user_instance.id])
+        resp = self.client.get(url)
+        self.assertEqual(resp.status_code, 200)
+        json_resp = json.loads(resp.content)
+        self.assertDictContainsSubset(model_to_dict(user_instance, MODEL_FIELDS), json_resp, 'it should give detail for the user asked with id')
+
+    def test_user_login(self):
+        self.create_user()
+        login_resp = self.client.post(
+            '/api/token', {'username': test_user['username'], 'password': test_user['password']})
+        self.assertEqual(login_resp.status_code, 200, 'it should login')
+
+    def test_loggedin_user_detail_view_without_token(self):
+        resp = self.client.get('/api/user/')
+        self.assertEqual(resp.status_code, 401, 'it should not give any detail of user')
+
+    def test_loggedin_user_detail_view(self):
+        user_instance = self.create_user()
+        login_resp = self.client.post(
+            '/api/token', {'username': test_user['username'], 'password': test_user['password']})
+        login_json_resp = json.loads(login_resp.content)
+        token = login_json_resp['access']
+        resp = self.client.get(
+            '/api/user/', HTTP_AUTHORIZATION='Bearer ' + token)
+        self.assertEqual(resp.status_code, 200)
+        json_resp = json.loads(resp.content)
+        self.assertDictContainsSubset(model_to_dict(user_instance, MODEL_FIELDS), json_resp, 'it should give detail for logged in user')
+
+    def test_user_register_view(self):
+        user_instance = self.create_user()
+        resp = self.client.post('/api/register/',
```

Comment:
what happens when you miss a field?

Reply:
If we miss any field the test will failed prompting with, the field is required.